consensus/src/chained_bft/event_processor.rs

```diff
@@ -54,6 +52,39 @@ mod event_processor_test;
 #[path = "event_processor_fuzzing.rs"]
 pub mod event_processor_fuzzing;
 
+struct SafetyRulesWrapper(SafetyRules);
+
+impl SafetyRulesWrapper {
+    pub fn consensus_state(&self) -> ConsensusState {
+        self.0.consensus_state()
+    }
+
+    pub fn update(&mut self, qc: &QuorumCert) {
```

Comment:
we probably can just return the updated state instead of using a wrapper, VoteInfo already carries consensus_state.

Reply:
Wouldn't that cause us to bump the counter in strange ways? An attack on our counters would be to just spam it blocks or a QC to trigger the counter updates. Not sure if that is okay.

Also I imagine there may be some state we want to keep somewhere between the safety rules and event processor and this gives us an abstract to play around in... I imagine eventually it could likely be dropped.